samples/common/src/utils.rs

```diff
@@ -116,75 +118,76 @@ pub async fn discover_digital_twin_provider_using_ibeji(
+    let mut client = ServiceRegistryClient::connect(chariott_uri.to_string())
         .await
         .map_err(|e| Status::internal(e.to_string()))?;
 
-    let request = Request::new(FulfillRequest {
-        namespace: constants::chariott::NAMESPACE_FOR_IBEJI.to_string(),
-        intent: Some(IntentMessage { intent: Some(IntentEnum::Discover(DiscoverIntent {})) }),
+    let request = Request::new(DiscoverRequest {
+        namespace: namespace.to_string(),
+        name: name.to_string(),
+        version: version.to_string(),
     });
 
-    // Get list of services at the requested namespace, if any.
-    let fulfillment_result: Option<Vec<discover_fulfillment::Service>> = client
-        .fulfill(request)
-        .await?
-        .into_inner()
-        .fulfillment
-        .and_then(|fulfillment_message| fulfillment_message.fulfillment)
-        .and_then(|fulfillment_enum| match fulfillment_enum {
-            FulfillmentEnum::Discover(discover) => Some(discover.services.into_iter().collect()),
-            _ => None,
-        });
-
-    // If we discovered one or more service, then return the URL for the first one that uses gRPC.
-    match fulfillment_result {
-        Some(services) =>
-            services.iter()
-                .find(|service| service.schema_kind == constants::chariott::SCHEMA_KIND_FOR_GRPC)
-                .map(|service| service.url.clone())
-                .ok_or_else(|| Status::not_found("Failed to discover the in-vehicle digital twin service's URL, as none of the services found had the '{constants::chariott::SCHEMA_KIND_FOR_GRPC}' schema kind")),
-        None => Err(Status::not_found("Failed to discover the in-vehicle digital twin service's URL, as it is not registered with Chariott"))
+    let response =
+        client.discover(request).await.map_err(|error| Status::internal(error.to_string()))?;
+    let response_inner = response.into_inner();
+
+    let service = response_inner.service;
+    if service.is_none() {
+        return Err(Status::not_found("Did not find the service"));
     }
+    let uri = service.unwrap().uri;
+
+    Ok(uri)
 }
 
-/// Retrieve the In-Vehicle Digital Twin URL.
-/// If invehicle_digital_twin_url is provided, then it's value is returned.
-/// Otherwise, chariott_url is used to retrieve it from Chariott.
+/// Retrieve the In-Vehicle Digital Twin URI.
+/// If invehicle_digital_twin_uri is provided, then it's value is returned.
+/// Otherwise, chariott_uri is used to retrieve it from Chariott.
 ///
 /// # Arguments
-/// * `invehicle_digital_twin_url` - Optional, In-Vehicle Digital Twin URL.
-/// * `chariott_url` - Optional, Chariott URL.
-pub async fn retrieve_invehicle_digital_twin_url(
-    invehicle_digital_twin_url: Option<String>,
-    chariott_url: Option<String>,
+/// * `invehicle_digital_twin_uri` - Optional, In-Vehicle Digital Twin URI.
+/// * `chariott_uri` - Optional, Chariott URI.
+pub async fn retrieve_invehicle_digital_twin_uri(
```

Comment:
Please consider my comment on  the `samples/common/src/consumer_config.rs` file about changing the Settings struct to an enum. 

I think this function will be cleaner as you'll only be passing one argument which would be a variant of the Settings enum, and you can match on its variants instead of the Option enum.

Reply:
This comes down to personal preferences.  Leaving it as is.